### PR TITLE
feat(items): stub in items

### PR DESCRIPTION
**Pull request by pahimar**
Adds lightweight item classes, basic item models, and localizations

### CODE REVIEW

File: src/main/java/com/pahimar/ee/init/ModItems.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+/**
+ * TODO Finish Javadoc
+ *
+ * @author      pahimar
+ *
+ * @since       3.0.0
+ */
+@GameRegistry.ObjectHolder(EquivalentExchange.MOD_ID)
+@Mod.EventBusSubscriber
+public class ModItems {
```
Comment (jamierocks) [inline src/main/java/com/pahimar/ee/init/ModItems.java]: If this is going to be a holder class, it should have the `final` modifier.

File: src/main/java/com/pahimar/ee/creativetab/CreativeTab.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * @since       3.0.0
+ */
+public class CreativeTab {
+
+    /**
+     * TODO Finish Javadoc
+     *
+     * @since   3.0.0
+     */
+    public static final CreativeTabs MOD_TAB = new CreativeTabs(EquivalentExchange.MOD_ID) {
```
Comment (jamierocks) [inline src/main/java/com/pahimar/ee/creativetab/CreativeTab.java]: IMO this constant is intrinsically linked with the mod (since there is only one creative tab), it would be better served placed in the `EquivalentExchange` class.

File: src/main/java/com/pahimar/ee/item/base/ItemBase.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+     *
+     * @since   3.0.0
+     */
+    @Override
+    @SideOnly(Side.CLIENT)
+    public void getSubItems(CreativeTabs tab, NonNullList<ItemStack> items) {
+        if (this.isInCreativeTab(tab)) {
+            if (getHasSubtypes() && variants.length > 0) {
+                for (int meta = 0; meta < variants.length; ++meta) {
+                    items.add(new ItemStack(this, 1, meta));
```
Comment (jamierocks) [inline src/main/java/com/pahimar/ee/item/base/ItemBase.java]: This is not the way things are done anymore, and won't be possible in 1.13 (goodbye item meta/damage values).

*See https://www.reddit.com/user/Dinnerbone/comments/6l6e3d/a_completely_incomplete_super_early_preview_of/*.

File: src/main/resources/assets/ee/lang/en_US.lang [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+item.ee:mercurial_eye.name=Mercurial Eye
+item.ee:minium_stone.name=Minium Stone
+item.ee:philosophers_stone.name=Philosophers Stone
+item.ee:watch.name=Watch of Flowing Time
+
+# Weapons
+item.ee:dark_matter_arrow.name=Dark Matter Arrow
+item.ee:dark_matter_bow.name=Dark Matter Bow
+item.ee:dark_matter_sword.name=Dark Matter Sword
+item.ee:kunai.name=Kunai
```
Comment (jamierocks) [inline src/main/resources/assets/ee/lang/en_US.lang]: Should probably introduce a newline here, seeing as its the standard everywhere else.

File: src/main/java/com/pahimar/ee/init/ModItems.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    /**
+     * TODO Finish Javadoc
+     */
+    private static final List<ItemBase> ITEMS = new ArrayList<>();
+
+    // Misc
+    public static final ItemBase ALCHEMICAL_BAG = new ItemAlchemicalBag();
```
Comment (jamierocks) [inline src/main/java/com/pahimar/ee/init/ModItems.java]: This isn't really standard use of the `GameRegistry.ObjectHolder` annotation, which as described by its javadoc **populates public static final fields with entries from the registry**.

### See Also

- MinecraftForge itself has a self-contained test class that's shows typical use of the new registry system (https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/src/test/java/net/minecraftforge/debug/ObjectHolderTest.java)
- Elytra's Smores is a good demo of the system (https://github.com/elytra/Smores/blob/1.12/src/main/java/com/elytradev/smores/Smores.java#L90, https://github.com/elytra/Smores/blob/1.12/src/main/java/com/elytradev/smores/init/SmoresBlocks.java#L37)
- cpw's simplepower mod, like Forge's test class, has a self contained example. (https://github.com/cpw/simplepower/blob/master/src/main/java/cpw/mods/simplepower/Objects.java#L35)

### tl;dr

The fields in a class annotated with `GameRegistry.ObjectHolder` need to match their (typically lowercase) resource name, and should be given a value of `null`. The items/blocks then need to be registered elsewhere (see the examples I linked above).

File: src/main/java/com/pahimar/ee/creativetab/CreativeTab.java (hunk 2 above)
Comment (pahimar) [reply src/main/java/com/pahimar/ee/creativetab/CreativeTab.java]: While it could be placed in the EquivalentExchange mod class, it's placed here to mimic (like all of my package structure) how Forge structures MC's package structure. It may seem superfluous to have a class containing a single constant such as this, but there may be mod developers that require multiple creative tabs and as such I went this way as it's extendable.

tldr; it's just style

File: src/main/java/com/pahimar/ee/init/ModItems.java (hunk 1 above)
Comment (pahimar) [reply src/main/java/com/pahimar/ee/init/ModItems.java]: Fair

File: src/main/resources/assets/ee/lang/en_US.lang (hunk 4 above)
Comment (pahimar) [reply src/main/resources/assets/ee/lang/en_US.lang]: If it comes to mind next time I'm working on the localization/model files I"ll see about leaving a newline at the end of each file, but I'll be honest it's not high on the priority list (don't get mad if I forget!) 😄

File: src/main/java/com/pahimar/ee/item/base/ItemBase.java (hunk 3 above)
Comment (pahimar) [reply src/main/java/com/pahimar/ee/item/base/ItemBase.java]: It's currently done in a similar way in Vanilla for 1.12 (check ItemDye), and yes it's going away in 1.13 but it is not 1.13 yet 😛

It's done in this way so that I have an easy generic way to add all variants of my items to the creative tab. It greatly simplifies constructors for items with subtypes (see https://github.com/pahimar/Equivalent-Exchange/blob/develop/1.12/src/main/java/com/pahimar/ee/item/ItemAlchemicalFuel.java#L20-L21)

File: src/main/java/com/pahimar/ee/init/ModItems.java (hunk 5 above)
Comment (pahimar) [reply src/main/java/com/pahimar/ee/init/ModItems.java]: Ah, interesting change in usage of the `ObjectHolder` annotation. I'll give the changes a whirl and see what I think - thanks for the tip
